cli/azd/pkg/project/framework_service_dotnet.go

```diff
@@ -41,16 +41,6 @@ func (dp *dotnetProject) RequiredExternalTools(context.Context) []tools.External
 
 // Initializes the docker project
 func (dp *dotnetProject) Initialize(ctx context.Context, serviceConfig *ServiceConfig) error {
-	if err := dp.dotnetCli.InitializeSecret(ctx, serviceConfig.Path()); err != nil {
```

Comment:
FWIW - We need to think about the implications of removing this (I'd love if we were able to, it does feel wrong in some way).  I think the reason we put these in .NET Secrets was it was the easiest way to ensure they would be available to the application when running locally (so it could find external resources).  I suspect there's a more idiomatic .NET way to do that (maybe we should be modifying/generating an appsettings file?) that doesn't look like `dotnet secrets`

Reply:
I haven't gotten that far yet. Trying to see what breaks with the change.

However, when running locally, .NET should behave as any 12-th factor app should behave, looking from environment variables. The `tasks.json` or launcher should be injecting environment variables. Maybe we also need `azd exec`?